packages/react-components/react-persona/src/components/Persona/usePersona.ts

```diff
@@ -76,6 +80,13 @@ export const usePersona_unstable = (props: PersonaProps, ref: React.Ref<HTMLElem
       })
     : undefined;
 
+  // When there's a PresenceBadge and and Avatar, the presence will be displayed by the Avatar. This means
+  // usePersonaStyles will not add persona's presence className. This ensures it will be added to the Avatar's badge.
+  if (props.presence && !presenceOnly) {
+    const avatarBadge = avatar?.badge as PresenceBadgeProps;
+    avatarBadge.className = mergeClasses(avatarBadge.className, personaClassNames.presence);
+  }
+
```

Comment:
It might be better to leave this out. It's not technically the Persona's presence slot, and the Avatar's badge slot already adds a static class name.